src/Microsoft.Framework.Runtime/ExportProviders/Resources/CreateCSharpManifestResourceName.cs

```diff
@@ -175,5 +178,19 @@ private static bool IsValidEverettIdChar(char c)
                 cat == UnicodeCategory.SpacingCombiningMark ||
                 cat == UnicodeCategory.EnclosingMark;
         }
+
+        public static string EnsureResourceExtension(string logicalName, string resourceFilePath)
+        {
+            string resourceExtension = Path.GetExtension(resourceFilePath);
+            if (!string.IsNullOrEmpty(resourceExtension))
+            {
+                if (!logicalName.EndsWith(resourceExtension, StringComparison.Ordinal))
```

Comment:
Ordinal. We always check the final name, including the extension, case sensitive